static/panes/device-view.ts

```diff
@@ -0,0 +1,378 @@
+                readOnly: true,
+                glyphMargin: true,
+                lineNumbersMinChars: 3,
+            })
+        );
+    }
+
+    override registerOpeningAnalyticsEvent(): void {
+        ga.proxy('send', {
+            hitType: 'event',
+            eventCategory: 'OpenViewPane',
+            eventAction: 'DeviceAsm',
+        });
+    }
+
+    initEditorActions(): void {
+        this.editor.addAction({
+            id: 'viewsource',
+            label: 'Scroll to source',
+            keybindings: [monaco.KeyMod.CtrlCmd | monaco.KeyCode.F10],
+            keybindingContext: undefined,
+            contextMenuGroupId: 'navigation',
+            contextMenuOrder: 1.5,
+            run: ed => {
+                const position = ed.getPosition();
+                if (position != null) {
+                    const desiredLine = position.lineNumber - 1;
+                    const source = this.deviceCode[desiredLine].source;
+                    if (source && source.file == null && this.compilerInfo.editorId != null) {
+                        // a null file means it was the user's source
+                        this.eventHub.emit('editorLinkLine', this.compilerInfo.editorId, source.line, -1, -1, true);
+                    }
+                }
+            },
+        });
+    }
+
+    initButtons(state: DeviceAsmState & MonacoPaneState): void {
+        this.fontScale = new FontScale(this.domRoot, state, this.editor);
+
+        this.topBar = this.domRoot.find('.top-bar');
+    }
+
+    override registerCallbacks(): void {
+        this.linkedFadeTimeoutId = null;
+        const mouseMoveThrottledFunction = _.throttle(this.onMouseMove.bind(this), 50);
+        this.editor.onMouseMove(e => mouseMoveThrottledFunction(e));
+
+        const cursorSelectionThrottledFunction = _.throttle(this.onDidChangeCursorSelection.bind(this), 500);
+        this.editor.onDidChangeCursorSelection(e => cursorSelectionThrottledFunction(e));
+
+        this.fontScale.on('change', this.updateState.bind(this));
+        this.selectize.on('change', this.onDeviceSelect.bind(this));
+        this.paneRenaming.on('renamePane', this.updateState.bind(this));
+
+        this.container.on('destroy', this.close, this);
+
+        this.eventHub.on('compileResult', this.onCompileResult, this);
+        this.eventHub.on('compiler', this.onCompiler, this);
+        this.eventHub.on('colours', this.onColours, this);
+        this.eventHub.on('panesLinkLine', this.onPanesLinkLine, this);
+        this.eventHub.on('compilerClose', this.onCompilerClose, this);
+        this.eventHub.on('settingsChange', this.onSettingsChange, this);
+        this.eventHub.emit('deviceViewOpened', this.compilerInfo.compilerId);
+        this.eventHub.emit('requestSettings');
+
+        this.container.on('resize', this.resize, this);
+        this.container.on('shown', this.resize, this);
+    }
+
+    override onCompileResult(id: number, compiler: any, result: any): void {
```

Comment:
The types for most `on...` callbacks are available in the `static/event-map.ts` file

Reply:
The types I declared as `any` are `unknown` in `static/event-map.ts`

Reply:
Oh then it's my bad! Sorry :)

Fixed in #4135 so you can update them :)